packages/thirdweb/src/contract/actions/resolve-abi.ts

```diff
@@ -1,9 +1,10 @@
 import { type Abi, formatAbi, parseAbi } from "abitype";
 import { download } from "../../storage/download.js";
+import { LruMap } from "../../utils/caching/lru.js";
 import { getClientFetch } from "../../utils/fetch.js";
 import { type ThirdwebContract, getContract } from "../contract.js";
 
-const ABI_RESOLUTION_CACHE = new WeakMap<ThirdwebContract<Abi>, Promise<Abi>>();
+const ABI_RESOLUTION_CACHE = new LruMap<Promise<Abi>>(1000);
```

Comment:
1000 might be too big here imo - abi can be rather large

also we need to make sure we do not cache `0x` responses - not sure if that is already addressed as part of the existing code (because it didn't matter as much before as it does now)

Reply:
I'd go with like `256` ish as the max size, seems big enough for almost all cases?

Reply:
done